### PR TITLE
feat(Typescript API client): Add initial version

### PR DESCRIPTION
**Pull request by nokome**
Adds a Typescript client for the Hub's public API. Intended for use in clients such as `gsuita` and `stencila` CLI.

### CODE REVIEW

**Review by alex-ketch — APPROVED** [review]
Thanks @nokome, this seems very useful and will greatly speed things up in the other projects.
I know we've had this conversation in other projects, but was wondering what you thought about not committing the generated files to the repo to reduce noise during PR reviews and searches?

**Comment by alex-ketch** [comment]
Oh, another observation: Noticed that this PR doesn't include our usual semantic release configuration, do you wish to release this package manually for now?

File: clients/typescript/src/models/InlineObject4.ts [hunk 1]
```diff
@@ -0,0 +1,335 @@
...
+ *
+ * This file is auto generated by OpenAPI Generator. Do not edit manually.
+ */
+import { exists, mapValues } from '../runtime';
+/**
+ * 
+ * @export
+ * @interface InlineObject4
+ */
+export interface InlineObject4 {
```
[the remark this answers is not in the document]
Comment (nokome) [reply clients/typescript/src/models/InlineObject4.ts:15]: Agreed, this also affects the Python client and needs changes in the `manager` API docs from which these get generated. For my reference, https://mux.com/blog/an-adventure-in-openapi-v3-api-code-generation/

**Comment by nokome** [comment]
> I know we've had this conversation in other projects, but was wondering what you thought about not committing the generated files to the repo to reduce noise during PR reviews and searches?

> Oh, another observation: Noticed that this PR doesn't include our usual semantic release configuration, do you wish to release this package manually for now?

These two are related. I was waiting to see if you felt this was a good idea before making it a NPM package that would get semantically released. If we do that, then yes, we do not need to commit the code.

The same goes for the Python client.

**Pull request merged**